src/ota_stm32h7.c

```diff
@@ -121,6 +124,12 @@ MG_IRAM static bool mg_stm32h7_swap(void) {
   flash_clear_err(bank);
   // printf("OPTSR_PRG 1 %#lx\n", FLASH->OPTSR_PRG);
   MG_SET_BITS(MG_REG(bank + FLASH_OPTSR_PRG), MG_BIT(31), desired);
+  if (STM_DEV_ID == 0x450) {
```

Comment:
Looks like  H745/H755 and H747/H757 aren't the only ones that use this ID. H743 uses 0x450 as well. I did not expect that, since H743 has a different RM. They shouldn't have allowed to have the same ID, it complicates things. How are we supposed to distinguish between a H743 and a H745 at runtime??

Reply:
We can always do it at compile time, but let me dive into ST waters and let's see what I can find.

Reply:
I know. It would be cleaner to do it at runtime though.

Reply:
Looks like single-core chips are in fact dual-core chips where the M4 has been disabled. This means that everything is the same, but probably we can check the BCM bits in the OPTION bytes.
Whether the chip is a single-core or a dual-core one, I think it is safe to say that if the CM4 is gated (no clock, BCM4 = 0 ), we can assume all flash is available.
WDYT ?
745 RM 13.3.14 SYSCFG_UR1
742 RM 12.3.12 shows this register is not present, notice the next register is UR2. The less-significant 16-bit portion of UR4 is reserved.
Worst case: we get a HardFault when accessing SYSCFG_UR1, this is an invasive way to detect a single core...
Best case: the option byte is readable

I don't see any other option if 743s really are 745s with the M4 disabled

Reply:
Good idea, but unfortunately UR1 reads as 0x10001 on both H743 and H745/H755... So no hard faults, but instead it returns the wrong value. It should have been 0x10000 instead.

I guess we'll have to decide at compile time if it's dual core or not. I did not find any other options.

Reply:
So 743s are in fact 745 chips with either a "broken" or removed M4 core...
The other options I can come up with, require collaboration between cores and a timeout... a big no no no...
Let's add a #define for all dual cores. Does this very same situation replicate with the other dual core families we handle ?

Reply:
Yes, it's the same for all dual core boards we have (H745/H755 and H747/H757). They are all under the same reference manual, anyway.

Reply:
So, I would add MG_OTA_H7DUALCORE or something like that (give or take one or two underscores)